### PR TITLE
Add streaming tests to Storage Write API Python wrapper

### PR DESCRIPTION
**Pull request by ahmedabu98**

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
R: @chamikaramj

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by Abacn** [comment]
Some tips to make tests running faster

- use wheels for sdk_location (like #25970 #26473), this would save 3.5 min per test
- run multiple tests in parallel by xdist option of pytest, like what is already doing for python postcommit

**Comment by Abacn** [comment]
Adding the flag is not yet effective as the sdk_location is set here for this test:

https://github.com/apache/beam/blob/4a09ea94727120596750edb8ce0127c12059c692/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L2489

instead of `needsSdkLocation`, probably provide the wheel location here and config that the test suite depends on initializeForDataflowJob.

https://github.com/apache/beam/blob/f5f7a471321c903174b452a1982c5183a79ac6bc/sdks/python/test-suites/dataflow/common.gradle#L457

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -2510,6 +2501,9 @@ class BeamModulePlugin implements Plugin<Project> {
           }
         }
       }
+      if (config.pythonPipelineOptions.contains("--runner=TestDataflowRunner")) {
+        pythonTask.configure {dependsOn ':sdks:python:test-suites:dataflow:initializeForDataflowJob'}
```
Comment (Abacn) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2505]: This actually won't work. `initializeForDataflowJob` exists in project like `:sdks:python:test-suites:dataflow:py310`.

We can add a parameter(s) `setupTasks` to `createCrossLanguageUsingJavaExpansionTask` , and then declare here

```
pythonTask.configure {dependsOn setupTask}
```

and call
```
createCrossLanguageUsingJavaExpansionTask(setupTask: 'initializeForDataflowJob'
...
```

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
if (config.pythonPipelineOptions.contains("--runner=TestDataflowRunner")) {
        setupTask.configure {dependsOn ":sdks:python:test-suites:dataflow:py" + project.ext.pythonVersion + ":initializeForDataflowJob"}
      }

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

File: sdks/python/apache_beam/io/external/xlang_bigqueryio_it_test.py [hunk 2]
```diff
@@ -234,6 +244,43 @@ def test_storage_write_beam_rows(self):
...
+          | beam.Map(lambda x: elements[x])
+          | beam.io.WriteToBigQuery(
+              table=table_id,
+              method=beam.io.WriteToBigQuery.Method.STORAGE_WRITE_API,
+              schema=schema,
+              with_auto_sharding=auto_sharding,
+              expansion_service=self.expansion_service))
+    hamcrest_assert(p, bq_matcher)
+
+  def test_streaming(self):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/external/xlang_bigqueryio_it_test.py:276]: May be let's also add a streaming test with "at_least_once" set.

**Review by chamikaramj — COMMENTED** [review]
Thanks.

**Comment by chamikaramj** [comment]
LGTM.

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

File: sdks/python/apache_beam/io/external/xlang_bigqueryio_it_test.py (hunk 2 above)
Comment (ahmedabu98) [reply sdks/python/apache_beam/io/external/xlang_bigqueryio_it_test.py:278]: Done

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run CommunityMetrics PreCommit

**Comment by ahmedabu98** [comment]
Run Website PreCommit

**Review by Abacn — APPROVED** [review]
LGTM

possible followup: move `CROSS_LANGUAGE_VALIDATES_RUNNER_PYTHON_VERSIONS` and the logic of pick up two py versions from jenkins script to gradle property of test-suites, 

https://github.com/apache/beam/blob/dc30afb959115097e18c596bcc8ef978fa5ca3fd/sdks/python/test-suites/gradle.properties#L36

Then claim a gradle task in dataflow/build.gradle: 

https://github.com/apache/beam/blob/dc30afb959115097e18c596bcc8ef978fa5ca3fd/sdks/python/test-suites/dataflow/build.gradle#L64

like what we did for validate runner tests:

The running time of the dataflow test suite would decrease by half because now py37 and py311 tests are running concurrently. Same thing can be done for the direct runner test.

**Pull request merged**